### PR TITLE
US-3.13.14:  Add and delete comment in task drawer

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskComments.vue [hunk 1]
```diff
@@ -40,69 +79,160 @@
...
+                } else if (elapsed < msPerMonth) {
+                    return this.$t('homepage.task.timeConvert.About_?_Days').replace('{0}', Math.round(elapsed / msPerDay));
+                } else if (elapsed === msPerMonth) {
+                    return this.$t('homepage.task.timeConvert.About_A_Month');
+                } else {
+                    return this.$t('homepage.task.timeConvert.About_?_Months').replace('{0}', Math.round(elapsed / msPerMonth));
+                }
             }
-          },
+            ,
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskComments.vue]: Not needed I think

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue [hunk 2]
```diff
@@ -181,7 +203,7 @@
   import LogDetails from './LogDetails.vue'
   import TaskComments from './TaskComments.vue'
 
-  import {updateTask, getTaskLogs, getTaskComments} from '../tasksAPI';
+  import {updateTask, getTaskLogs, getTaskComments,addTaskComments} from '../tasksAPI';
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]:
```suggestion
  import {updateTask, getTaskLogs, getTaskComments, addTaskComments} from '../tasksAPI';
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue [hunk 3]
```diff
@@ -220,12 +246,24 @@
...
+    },
     watch: {
       'task.description': function(newValue, oldValue) {  
         if (newValue !== oldValue) { 
           this.autoSaveDescription(); 
         } 
-      }
+      },
+      editorData(val) {
+        if (val=== '') {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]: this.disabledComment = val === '';

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue [hunk 4]
```diff
@@ -199,6 +221,10 @@
     },
     data() {
       return {
+        editorData:'',
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]:
```suggestion
        editorData: '',
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue [hunk 5]
```diff
@@ -199,6 +221,10 @@
...
+        emptyValue:'',
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]:
```suggestion
        emptyValue: '',
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue [hunk 6]
```diff
@@ -199,6 +221,10 @@
...
+        reset:false,
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]:
```suggestion
        reset: false,
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue [hunk 7]
```diff
@@ -199,6 +221,10 @@
...
+        disabledComment : true,
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]:
```suggestion
        disabledComment: true,
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js [hunk 8]
```diff
@@ -185,3 +185,56 @@ export function getTaskComments(taskId) {
...
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(comment)
+  }).then((resp) => {
+    if(resp && resp.ok) {
+      return resp.json();
+    }
+    else {
+      throw new Error ('Error when adding comment');
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js]:
```suggestion
      throw new Error ('Error when adding task sub comment');
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js [hunk 9]
```diff
@@ -185,3 +185,56 @@ export function getTaskComments(taskId) {
...
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js]:
```suggestion
      throw new Error ('Error when adding task comment');
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js [hunk 10]
```diff
@@ -185,3 +185,56 @@ export function getTaskComments(taskId) {
...
+    }
+  })
+}
+
+export function removeComment(commentId) {
```
Comment (azayati) [inline digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js]:
```suggestion
export function removeTaskComment(commentId) {
```

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskComments.vue (hunk 1 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskComments.vue]: removed

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue (hunk 2 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue (hunk 3 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue (hunk 4 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue (hunk 5 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue (hunk 6 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue (hunk 7 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/components/TaskDrawer.vue]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js (hunk 9 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js]: Done

File: digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js (hunk 10 above)
Comment (AzmiTouil) [reply digital-workplace-webapps/src/main/webapp/vue-app/tasks/tasksAPI.js]: Done

**Pull request merged**